Jenkinsfile

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env groovy
+
+@Library('apm@v1.0.6') _
+
+import co.elastic.matrix.*
+import groovy.transform.Field
+
+@Field stackVersions = ['6.5.0', '6.6.0', '7.0.0-alpha1-SNAPSHOT']
+@Field scopes = ['elastic-apm-js-core', 'elastic-apm-js-base']
+
+/**
+This is the parallel tasks generator,
+it is need as field to store the results of the tests.
+*/
+@Field def rumTasksGen
+
+pipeline {
+  agent any
+  environment {
+    BASE_DIR="src/github.com/elastic/apm-agent-js-base"
+    NOTIFY_TO = credentials('notify-to')
+    JOB_GCS_BUCKET = credentials('gcs-bucket')
+    PIPELINE_LOG_LEVEL='INFO'
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  triggers {
+    issueCommentTrigger('.*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
+  }
+  parameters {
+    booleanParam(name: 'Run_As_Master_Branch', defaultValue: false, description: 'Allow to run any steps on a PR, some steps normally only run on master branch.')
+    booleanParam(name: 'doc_ci', defaultValue: true, description: 'Enable build docs.')
+  }
+  stages {
+    stage('Initializing'){
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+      }
+      stages {
+        /**
+        Checkout the code and stash it, to use it on other stages.
+        */
+        stage('Checkout') {
+          steps {
+            deleteDir()
+            gitCheckout(basedir: "${BASE_DIR}")
+            stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+          }
+        }
+        /**
+        Lint the code.
+        */
+        stage('Lint') {
+          steps {
+            deleteDir()
+            unstash 'source'
+            script{
+              docker.image('node:8-alpine').inside(){
+                dir("${BASE_DIR}"){
+                  sh(label: "Lint", script: '''
+                  HOME=$(pwd)
+                  npm install get-dependencies --save
+                  npm run lint
+                  ''')
+                }
+                stash allowEmpty: true, name: 'cache', includes: "${BASE_DIR}/.npm/**", useDefaultExcludes: false
+              }
+            }
+          }
+        }
+        /**
+        Execute unit tests.
+        */
+        stage('Test') {
+          steps {
+            deleteDir()
+            runParallelTest()
+          }
+        }
+        /**
+        Build the documentation.
+        */
+        stage('Documentation') {
+          when {
+            beforeAgent true
+            allOf {
+              anyOf {
+                not {
+                  changeRequest()
+                }
+                branch 'master'
```

Comment:
@jahtalab Pointed out that it will be good to have docs build for all branches.

Reply:
I will add it on files changes inside the docs folder, in other cases, it does not have sense.

Reply:
I think we need to test docs PRs before merging them, which means we need to run tests on all branches. I'm open to suggestions about how to test doc PRs in a more efficient way.